packages/cubejs-client-react/index.d.ts

```diff
@@ -477,25 +483,47 @@ declare module '@cubejs-client/react' {
    * />
    * ```
    */
-  type MemberUpdater = {
-    add: (member: MemberType) => void;
-    remove: (member: MemberType) => void;
-    update: (member: MemberType, updateWith: MemberType) => void;
+
+  type FilterWithExtraFields = Omit<Filter, 'dimension'> & {
+    dimension: TCubeDimension | TCubeMeasure;
+    operators: { name: string; title: string }[];
+  };
+  type TimeDimensionWithExtraFields = Omit<TimeDimension, 'dimension'> & {
+    dimension: TCubeDimension & { granularities: { name: string; title: string }[] };
+  };
+
+  type MemberUpdater<T> = {
+    add: (member: T) => void;
+    remove: (member: { index: number }) => void;
+    update: (member: { index: number }, updateWith: T) => void;
   };
+  type DimensionUpdater = MemberUpdater<TCubeDimension>;
+  type MeasureUpdater = MemberUpdater<TCubeMeasure>;
+  type SegmentUpdater = MemberUpdater<TCubeSegment>;
+  // Only require the fields that are actually used (otherwise fields like `operators` are required just to add/update)
+  type TimeDimensionUpdater = MemberUpdater<
```

Comment:
Would it be possible to simplify this type? While it may be absolutely correct, we use them for docs generation (typedoc) and if we run it we'll get

```
TimeDimensionUpdater: MemberUpdater‹object | object & object›
```

It is ok to duplicate some types, as the main goal here is clear typedoc output.

Reply:
Got it, installing typedoc now so I can preview output and make it look pretty :)